### DISCUSSION TITLE
Getting ipynb to open as a notebook in GitHub codespace with openFiles?

### DISCUSSION THREAD

**Discussion by pamelafox** [microsoft/vscode-jupyter#13723] [Questions and Answers]
I am trying to create a dev container that automatically opens up a particular notebook file using the Jupyter extension. All of my attempts result in the notebook rendering as JSON instead of a notebook.

Example devcontainer.json:

```
{
	"name": "Data 8: Project 1 Notebook",
	"build": {
		"dockerfile": "../dockerfile",
		"context": "../.."
	},
	"waitFor": "onCreateCommand",
	"customizations": {
		"codespaces": {
			"openFiles": [
			  "materials/fa22/project/project1/project1.ipynb"
			]
		},
		"vscode": {
			"settings": {
				"python.defaultInterpreterPath": "/usr/local/bin/python",
				"workbench.editorAssociations": {
					"*.ipynb": "jupyter-notebook"
				},
				"notebook.compactView": false
			},
			"extensions": [
				"ms-python.python",
				"ms-toolsai.jupyter",
				"MS-vsliveshare.vsliveshare"
			]
		}
	}
}
```

For this repo:
https://github.com/pamelafox/materials-fa22

Here's another repo that demonstrates the issue. If you create a Codespace, you'll notice it opens 4 ipynb files, but they render as JSON, not notebooks.

https://github.com/janosh/pymatviz

If you then close them and re-open them, they open as expected, with the notebook UI.

I'm not sure if this is an extension issue or a Codespace issue, as I assume it's related to timing of the customizations and extensions. It'd be really fantastic if there was a way to make this work, since this use case would be very popular with data science teachers.

**Comment by DonJayamanne** [comment]
Thanks for posting this question
I was able to repro this issue, looks like an issue in codespace of VS Code Core.
Will transfer the issue to VS Code and have a look into this.

**Reply by pamelafox** [reply]
Thank you! If you find any workaround that'd work this week, that'd be great, as I'm presenting Jupyter notebooks in Codespaces at a data science for teachers conference. Otherwise we can try it again in the fall.

**Reply by pamelafox** [reply]
David Gardiner gave me a workaround for now: https://github.com/orgs/community/discussions/58399#discussioncomment-6222762
